utils/validations.tsx

```diff
@@ -1,97 +1,136 @@
 import { PublicKey } from '@solana/web3.js'
-import { TokenAccountInfo } from '@tools/validators/accounts/token'
 import { ProgramBufferAccount } from '@tools/validators/accounts/upgradeable-program'
 import { tryParseKey } from '@tools/validators/pubkey'
 import { create } from 'superstruct'
 import { tryGetTokenAccount } from './tokens'
 import * as yup from 'yup'
 import { getMintNaturalAmountFromDecimal } from '@tools/sdk/units'
 import { BN } from '@project-serum/anchor'
+import { ConnectionContext } from 'stores/useWalletStore'
+import {
+  AccountInfo,
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  Token,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-token'
+import { Connection } from '@solana/web3.js'
 
-export const validateDestinationAccAddress = async (
-  connection,
-  val: any,
-  governedAccount?: PublicKey
+const getValidateAccount = async (
+  connection: Connection,
+  pubKey: PublicKey
 ) => {
+  const account = await connection.getParsedAccountInfo(pubKey)
+  if (!account || !account.value) {
+    throw 'Account not found'
+  }
+  return account
+}
+
+const getValidatePublicKey = (val: string) => {
   const pubKey = tryParseKey(val)
   if (pubKey) {
-    const account = await connection.current.getParsedAccountInfo(pubKey)
-    if (!account || !account.value) {
-      throw 'Account not found'
-    }
-    if (
-      !(
-        'parsed' in account.value.data &&
-        account.value.data.program === 'spl-token'
-      )
-    ) {
+    return pubKey
+  } else {
+    throw 'Provided value is not a public key'
+  }
+}
+
+const isValidSplTokenAccount = (account, exceptionMode = false) => {
+  if (
+    !(
+      'parsed' in account.value.data &&
+      account.value.data.program === 'spl-token'
+    )
+  ) {
+    if (exceptionMode) {
       throw 'Invalid spl token account'
     }
+    return false
+  }
+  return true
+}
 
-    let tokenAccount
+const validateDoseTokenAccountMatchMint = (
+  tokenAccount: AccountInfo,
+  mint: PublicKey
+) => {
+  if (tokenAccount.mint.toBase58() !== mint.toBase58()) {
+    throw "Account mint doesn't match source account"
+  }
+}
 
-    try {
-      tokenAccount = create(account.value.data.parsed.info, TokenAccountInfo)
-    } catch {
-      throw 'Invalid spl token account'
-    }
+export const getDoseAtaExists = async (
```

Comment:
nit: `tryGetAta()` would be more consistent with other naming conventions used in the project